keras/callbacks.py

```diff
@@ -779,32 +785,46 @@ def is_indexed_slices(grad):
         else:
             self.writer = tf.summary.FileWriter(self.log_dir)
 
-        if self.embeddings_freq:
+        if self.embeddings_freq and self.embeddings_data is not None:
             embeddings_layer_names = self.embeddings_layer_names
 
             if not embeddings_layer_names:
                 embeddings_layer_names = [layer.name for layer in self.model.layers
                                           if type(layer).__name__ == 'Embedding']
+            self.embeddings = []
+            embeddings_vars = {}
 
-            embeddings = {layer.name: layer.weights[0]
-                          for layer in self.model.layers
-                          if layer.name in embeddings_layer_names}
+            for layer in self.model.layers:
+                if layer.name in embeddings_layer_names:
+
+                    self.batch_id = batch_id = tf.placeholder(tf.int32)
+                    self.step = step = tf.placeholder(tf.int32)
 
-            self.saver = tf.train.Saver(list(embeddings.values()))
+                    embedding_input = self.model.get_layer(layer.name).output
+                    embedding_size = reduce(mul, embedding_input.shape[1:])
+                    embedding_input = tf.reshape(embedding_input,
+                                                 (step, int(embedding_size)))
+                    shape = (self.embeddings_data.shape[0], int(embedding_size))
+                    embedding = tf.Variable(tf.zeros(shape),
```

Comment:
Is it possible to use Keras' backend here?

Reply:
It's probably possible but unnecessary because this is all inside the TensorBoard callback which only works with TF backend.

Reply:
I agree that it will only work with tensorflow but in terms of style it will improve the quality of your PR. It limits the use of tensorflow outside of the backend.

Reply:
Not sure what you mean by

> It limits the use of tensorflow outside of the backend.

Could you please rephrase or elaborate?

Why would you consider it better style to use the generic Keras backend although this code can never be used with any other backend then the TF backend?

Can we maybe have a third opinion on that, @chleibig ?

Reply:
The idea was: if we can use Keras' TF backend we should use it, if we cannot we fall back to native TF. This way we have less backend specific code using native TF outside of the Keras bakend (even if we can only use this callback only with the TF backend and for some parts we use native TF). It's more about style than usability. I would be the first time a `tf.Variable` is used outside of the backend.
Since we have to use tensorflow to write to the logs and we don't use the extra attributes given to the tensors it's not that important. But I would wait for someone else to comment on that.

Reply:
I totally agree with @tboquet that it would be better style but I would only do it if all `tf` statements can be avoided in this TF specific callback. Otherwise we still have a mixture of backend and native statements.

Reply:
We can't get rid of all the `tf`-statements because of the `FileWriter` and `Saver` and other `tf`-specific stuff which is not available on Keras' backend.